packages/entities/entities-plugins/docs/plugin-select.md

```diff
@@ -0,0 +1,190 @@
+- properties:
+  - `app`:
+    - type: `'konnect' | 'kongManager'`
+    - required: `true`
+    - default: `undefined`
+    - App name.
+
+  - `apiBaseUrl`:
+    - type: `string`
+    - required: `true`
+    - default: `undefined`
+    - Base URL for API requests.
+
+  - `requestHeaders`:
+    - type: `RawAxiosRequestHeaders | AxiosHeaders`
+    - required: `false`
+    - default: `undefined`
+    - Additional headers to send with all Axios requests.
+
+  - `getCreateRoute`:
+    - type: `(plugin: string) => RouteLocationRaw`
+    - required: `true`
+    - default: `undefined`
+    - A function that returns the route for creating a specific plugin type.
+
+  - `createCustomRoute`:
+    - type: RouteLocationRaw
+    - required: `false`
+    - default: `undefined`
+    - The route for creating a custom plugin.
+
+  - `getCustomEditRoute`:
+    - type: `(plugin: string) => RouteLocationRaw`
+    - required: `false`
+    - default: `undefined`
+    - A function that returns the route for editing a custom plugin.
+
+  - `workspace`:
+    - type: `string`
+    - required: `true`
+    - default: `undefined`
+    - *Specific to Kong Manager*. Name of the current workspace.
+
+  - `controlPlaneId`:
+    - type: `string`
+    - required: `true`
+    - default: `undefined`
+    - *Specific to Konnect*. Name of the current control plane.
+
+  - `entityId`:
+    - type: `string`
+    - required: `false`
+    - default: `null`
+    - Current entity id if the PluginSelect is being launched from the plugins tab on a consumer, consumer group, gateway service, or route detail page.
+
+  - `entityType`:
+    - type: `'consumers' | 'consumer_groups' | 'services' | 'routes'`
+    - required: `false`
+    - default: `null`
+    - Current entity type if the PluginSelect is being launched from the plugins tab on a consumer, consumer group, gateway service, or route detail page.
+
+The base konnect or kongManger config.
+
+#### `canCreateCustom`
+
+- type: `Function as PropType<() => boolean | Promise<boolean>>`
+- required: `false`
+- default: `async () => true`
+
+A synchronous or asynchronous function, that returns a boolean, that evaluates if the user can create a new custom plugin.
+
+#### `canDeleteCustom`
+
+- type: `Function as PropType<(row: object) => boolean | Promise<boolean>>`
+- required: `false`
+- default: `async () => true`
+
+A synchronous or asynchronous function, that returns a boolean, that evaluates if the user can delete a given custom plugin.
+
+#### `canEditCustom`
+
+- type: `Function as PropType<(row: object) => boolean | Promise<boolean>>`
+- required: `false`
+- default: `async () => true`
+
+A synchronous or asynchronous function, that returns a boolean, that evaluates if the user can edit a given custom plugin.
+
+#### `noRouteChange`
+
+- type: `boolean`
+- required: `false`
+- default: `false`
+
+If true, let consuming component handle event when clicking on a plugin. Used in conjunction with `@plugin-clicked` event.
```

Comment:
We can't do this because the handler being defined on `PluginSelect` is different than the handler defined on the `PluginSelectCard` component which is the component that actually evaluates what to do on click.

I will change the name.